doc/specification/language-proposal.md

```diff
@@ -0,0 +1,1518 @@
+
+The expression assigned to _α'_ in its field definition is the sum of terms _κᵢ_·Π*Lᵢ*·(*μ*(*α*; *Rᵢ*) - *μ*(*α*; *Lᵢ*)) for each reaction _i_.
+
+Example: consider the reaction system below involving three species _a_, _b_ and _c_, represented by their concentrations, coupled via a reaction rate to a quantity _x_ that is governed also by a simple ODE.
+
+```
+{
+    x' = 1.23 M⁻³s⁻²;
+    2a + b + c → 2b (x);
+    ∅ → c (3.4 M/s);
+}
+```
+
+The record literal is then algebraically equivalent to the following.
+
+```
+{
+    x' = 1.23 M⁻³s⁻²;
+    a' = -2·x·a²·b·c;
+    b' = x·a²·b·c;
+    c' = -x·a²·b·c + 3.4 M/s;
+}
+```
+
+### Functions
+<a id="functions"/>
+
+A function literal gives a value of a function type. Function types have no representation in the Arblang source language, and so cannot be used in type assertions, or in function arguments. Function literals may only appear in _function-application_ expressions or as the bound value in a _let-binding_ or module _function-defn_.
+
+> _function-literal_ ::= `fn` `(` ( _function-arg_ ( `,` _function-arg_ )* )? `)` ***right-arrow*** _expression_ | `(` _function-literal_ `)`
+>
+> _function-arg_ ::= ***symbol*** _type-assertion_
+
+The identifiers introduced by _function-arg_ clauses have function scope which comprises the final defining expression. These are bound to parameter values in a function call expression (see _Expressions_ below). The final expression may be of any non-function type, but may involve identifiers bound only to: other functions, _parameter-constant_ expressions, or the function arguments. In particular, the expression may not contain identifiers bound to an external _bindable_ in an interface.
+
+
+## Expressions
+<a id="expressions"/>
+
+The expression syntax below is ambiguous in that, for example, a function application or qualified identifier may be parsed as a _boolean-expr_, _algebraic-expr_, or _record-expr_. Each possible parse should be semantically equivalent.
+
+> _expression_ ::= ( _value-binding_ | _conditional-expr_ | _boolean-expr_ | _algebraic-expr_ | _record-expr_ ) _type-assertion_? | `(` _expression_ `)`
+
+An expression with type assertion of the form `expr: T` is valid if the type of `expr` is `T` or a supertype of `T`, and ill-formed otherwise. The type of `expr: T`, if well-formed, is always the type `T`.
+
+Each identifier or qualified identifier in an expression must be bound (in the context determined by its occurrence). In expression context, the identifier or qualified identifier may be bound to a function argument, to another expression via a value binding in an outer expression, to a record field, to a module constant or function definition, to a module parameter, or to an external quantity via an interface binding.
```

Comment:
A qualified identifier can't be bound directly to a function argument, but it can be bound to a record field, or a value in another module etc.

e.g.
```
module foo {
    def a = 3;
}
module bar {
    def x = 3 * foo.b; # invalid: foo.b not bound to anything.
    def y = 3 * foo.a; # ok: foo.a is bound to the value '3'.
}
```

Outer expression really means enclosing expression, and I should possibly  use that term instead. For example, `with S; expr` is the enclosing expression for the expression `expr`. Alternately, `expr` could be termed a subexpression of `with S; expr`.

Reply:
Okay, got it. Thanks!